portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -1906,6 +1906,15 @@ public void setTreePaths(long folderId, String treePath, boolean reindex)
 
 		indexableActionableDynamicQuery.setAddCriteriaMethod(
 			dynamicQuery -> {
+				DLFolder folder = _dlFolderLocalService.fetchDLFolder(folderId);
+
+				if (folder != null) {
+					Property groupIdProperty = PropertyFactoryUtil.forName(
+						"groupId");
+
+					dynamicQuery.add(groupIdProperty.eq(folder.getGroupId()));
+				}
+
```

Comment:
When can `folder` be null? If this is the root folder case, can you account for it explicitly instead of invoking the fetch method? E.g.
```
if (folderId != DLFolderConstants.DEFAULT_FOLDER_ID) {
  DLFolder dlFodler = _dlFolderLocalService.fetchDLFolder(folderId);
  ...
}
```